### PR TITLE
changed parseType Literal to EnumLiteral and added descriptions

### PR DESCRIPTION
**Pull request by colameo**

### CODE REVIEW

File: src/JDLApplicationOptions.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
     "values": [
       "primary"
     ],
-    "description": "You can put whatever value you want, provided you know it will work (like dark, or light), can also be empty"
```
Comment (colameo) [inline src/JDLApplicationOptions.json:82]: @MathieuAA note please that an option must have in the jhide always a value

File: JDLApplicationOptions.md [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
-| authenticationType   | STRING   | Literal               | jwt, uaa, session, oauth2                                                               | uaa for UAA apps,  jwt otherwise                                                                              |
+| clientPackageManager | STRING   | EnumLiteral           | npm, yarn                                                                             | One of the listed values or one of the prod database type                                                    |
+| databaseType         | STRING   | EnumLiteral           | sql, mongodb, cassandra, couchbase, no                                                   | One of the listed values or one of the prod database type                                                    |
+| uaaBaseName          | STRING   | EnumLiteral           | "uaa"                                                                                | Mandatory for gateway and microservices if auth type is uaa,  must be between double-quotes                   |
```
Comment (MathieuAA) [inline JDLApplicationOptions.md]: Here the default value is indeed "uaa", but a user can set one

**Comment by MathieuAA** [comment]
Everything except `uaaBaseName` looks good!

File: JDLApplicationOptions.md (hunk 2 above)
Comment (colameo) [reply JDLApplicationOptions.md]: had a look to application_options.js, is '**../uaa**' the right one?

Comment (MathieuAA) [reply JDLApplicationOptions.md]: I don't get what you mean, but I think changing it to a Literal should work

Comment (MathieuAA) [reply JDLApplicationOptions.md]: About the value, I think you're right

**Comment by colameo** [comment]
@MathieuAA can we merge this PR or did I miss something?

**Comment by MathieuAA** [comment]
Yup we can

**Review by MathieuAA — APPROVED** [review]

**Comment by MathieuAA** [comment]
Sorry I'm kinda busy for the WE

**Pull request merged**